src/internal_modules/roc_core/mpsc_queue_impl.cpp

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2023 Roc Streaming authors
```

Comment:
nitpick: since this file is mostly copied from another file, it'd be more correct to preserve original copyright year (because it's the year from which copyright for this code starts).

Reply:
That's my fault. Thanks.